pulpcore/cli/ansible/context.py

```diff
@@ -0,0 +1,59 @@
+from pulpcore.cli.common.context import (
+    PulpEntityContext,
+    PulpRepositoryContext,
+    PulpRepositoryVersionContext,
+)
+
+
+# TODO Add Role and Collection Content contexts
+
+
+class PulpAnsibleDistributionContext(PulpEntityContext):
+    ENTITY = "distribution"
+    HREF = "ansible_ansible_distribution_href"
+    LIST_ID = "distributions_ansible_ansible_list"
+    READ_ID = "distributions_ansible_ansible_read"
+    CREATE_ID = "distributions_ansible_ansible_create"
+    UPDATE_ID = "distributions_ansible_ansible_update"
+    DELETE_ID = "distributions_ansible_ansible_delete"
+
+
+class PulpAnsibleRoleRemoteContext(PulpEntityContext):
+    ENTITY = "role remote"
+    HREF = "ansible_role_remote_href"
+    LIST_ID = "remotes_ansible_role_list"
+    READ_ID = "remotes_ansible_role_read"
+    CREATE_ID = "remotes_ansible_role_create"
+    UPDATE_ID = "remotes_ansible_role_partial_update"  # This makes update have less requirements
```

Comment:
Please let's keep this consistent.
Maybe we should do PATCH instead of PUT in all places. It exhibits less danger for race conditions.
@ggainey @daviddavis @dkliban ?

Reply:
I agree that using PATCH is probably better than PUT but most places like here use async tasks to perform updates so it's not a high priority IMO.